src/main/frontend/app/stores/flow-store.ts

```diff
@@ -249,38 +252,48 @@ const useFlowStore = create<FlowState>((set, get) => ({
       }),
     })
   },
-  updateChild: (parentNodeId: string, updatedChild: ChildNode) => {
+  updateChild: (rootNodeId: string, updatedChild: ChildNode) => {
     set({
       nodes: get().nodes.map((node) => {
-        if (node.id === parentNodeId && isFrankNode(node)) {
-          const updatedChildren = node.data.children.map((child) =>
-            child.id === updatedChild.id ? { ...child, ...updatedChild } : child,
-          )
-          return {
-            ...node,
-            data: {
-              ...node.data,
-              children: updatedChildren,
-            },
-          }
+        if (node.id !== rootNodeId || !isFrankNode(node)) return node
+
+        return {
+          ...node,
+          data: {
+            ...node.data,
+            children: updateChildRecursive(node.data.children || [], updatedChild),
+          },
         }
-        return node
       }),
     })
   },
-  deleteChild: (parentId, childId) => {
+  deleteChild: (rootNodeId: string, childId: string) => {
     set({
       nodes: get().nodes.map((node) => {
-        if (node.id === parentId && isFrankNode(node)) {
-          return {
-            ...node,
-            data: {
-              ...node.data,
-              children: node.data.children.filter((child) => child.id !== childId),
-            },
-          }
+        if (node.id !== rootNodeId || !isFrankNode(node)) return node
+
+        return {
+          ...node,
+          data: {
+            ...node.data,
+            children: deleteChildRecursive(node.data.children || [], childId),
+          },
+        }
+      }),
+    })
+  },
+  addChildToChild: (nodeId: string, targetChildId: string, newChild: ChildNode) => {
+    set({
+      nodes: get().nodes.map((node) => {
+        if (node.id !== nodeId) return node
+
+        return {
+          ...node,
+          data: {
+            ...node.data,
+            children: addChildRecursive(node.data.children || [], targetChildId, newChild),
```

Comment:
Don't forget to use `??` instead of `||` in order to prevent falsely values triggering unwanted behaviour, I know that won't be the case here but its better to be consistent with it